### PR TITLE
refactor lowess smoothing

### PR DESCRIPTION
**Pull request by mathause**
- [x] Related #146 & #149
 - [x] Tests added
 - [x] Passes `isort . && black . && flake8`
 - [ ] Fully documented, including `CHANGELOG.rst`

Spitting out the lowess smoothing. This is much nicer and cleaner again.

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -26,6 +26,9 @@ New Features
 - Extract function to select the order of the auto regressive model: ``mesmer.core.auto_regression._select_ar_order_xr``
   (`#176 <https://github.com/MESMER-group/mesmer/pull/176>`_).
   By `Mathias Hauser <https://github.com/mathause>`_.
+- Refactor the LOWESS smooting for xarray objects: :py:func:`mesmer.core.smooting.lowess`.
```
Comment (znicholls) [inline CHANGELOG.rst:29]: `smooting` --> `smoothing`

File: docs/source/api.rst [hunk 2]
```diff
@@ -25,6 +25,7 @@ Statistical core functions
    ~core.auto_regression._select_ar_order_xr
    ~core.auto_regression._fit_auto_regression_xr
    ~core.auto_regression._draw_auto_regression_correlated_np
+   ~core.smooting.lowess
```
Comment (znicholls) [inline docs/source/api.rst:28]: `smooting` --> `smoothing`

**Review by znicholls — COMMENTED** [review]
Beauty! Small tweaks

File: CHANGELOG.rst (hunk 1 above)
Comment (mathause) [reply CHANGELOG.rst:29]: Thanks! Fixed in #194

**Pull request merged**